robot/impl/robot_reconfigure_test.go

```diff
@@ -1981,10 +1981,16 @@ func TestRobotReconfigure(t *testing.T) {
 
 		rr.triggerConfig <- struct{}{}
 
-		utils.SelectContextOrWait(context.Background(), 200*time.Millisecond)
-		mock6, err = robot.ResourceByName(mockNamed("mock6"))
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, mock6.(*mockFake).reconfCount, test.ShouldEqual, 1)
+		testutils.WaitForAssertionWithSleep(t, time.Millisecond*100, 30, func(tb testing.TB) {
+			mock6, err := robot.ResourceByName(mockNamed("mock6"))
+			test.That(tb, err, test.ShouldBeNil)
+			test.That(tb, mock6, test.ShouldNotBeNil)
+			// test failures don't abort in helper functions, so have to avoid nil pointer panic manually
```

Comment:
Might not need this if you don't have `Helper()`.

Reply:
As mentioned above, the WaitFor...  itself is the helper function. This was required rather or not I explicitly set tb.Helper() here. (It just changes the output a little bit around what line numbers it prints as failures.)